Cannot enable android with emulate doubles

### Godot version

https://github.com/godotengine/godot/commit/2e3662acbd8586697db38569785cc9f97dff945e

### System information

Windows 11, Nvidia 3000

### Issue description

The android platform cannot enable emulate double floats.

```
platform/android/jni_utils.cpp:271:11: error: cannot initialize a variable of type 'real_t *' (aka 'double *') with an rvalue of type 'float *'
build_rd_headers(["servers/rendering/renderer_rd/shaders/effects/copy_to_fb.glsl.gen.h"], ["servers/rendering/renderer_rd/shaders/effects/copy_to_fb.glsl"])
                real_t *w = sarr.ptrw();
                        ^   ~~~~~~~~~~~
platform/android/jni_utils.cpp:287:11: error: cannot initialize a variable of type 'real_t *' (aka 'double *') with an rvalue of type 'float *'
                real_t *w = sarr.ptrw();
                        ^   ~~~~~~~~~~~
```

### Steps to reproduce

See https://github.com/godotengine/godot/pull/67571

### Minimal reproduction project

None

--- a/platform/android/jni_utils.cpp
+++ b/platform/android/jni_utils.cpp
@@ -265,33 +265,33 @@ Variant _jobject_to_variant(JNIEnv *env, jobject obj) {
 	if (name == "[D") {
 		jdoubleArray arr = (jdoubleArray)obj;
 		int fCount = env->GetArrayLength(arr);
-		PackedFloat32Array sarr;
-		sarr.resize(fCount);
+		PackedFloat64Array packed_array;
+		packed_array.resize(fCount);
 
-		real_t *w = sarr.ptrw();
+		double *w = packed_array.ptrw();
 
 		for (int i = 0; i < fCount; i++) {
 			double n;
 			env->GetDoubleArrayRegion(arr, i, 1, &n);
 			w[i] = n;
 		}
-		return sarr;
+		return packed_array;
 	}
 
 	if (name == "[F") {
 		jfloatArray arr = (jfloatArray)obj;
 		int fCount = env->GetArrayLength(arr);
-		PackedFloat32Array sarr;
-		sarr.resize(fCount);
+		PackedFloat32Array packed_array;
+		packed_array.resize(fCount);
 
-		real_t *w = sarr.ptrw();
+		float *w = packed_array.ptrw();
 
 		for (int i = 0; i < fCount; i++) {
 			float n;
 			env->GetFloatArrayRegion(arr, i, 1, &n);
 			w[i] = n;
 		}
-		return sarr;
+		return packed_array;
 	}
 
 	if (name == "[Ljava.lang.Object;") {

